geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -11,23 +12,40 @@ export class ShapefileService {
     const outputKey = `shapefile-${new Date().getTime()}.geojson`;
 
     const _geoJson = await mapshaper.applyCommands(
-      `-i ${fileInfo.path.replace('.zip', '')}/*.shp -info -o ${outputKey}`,
+      `-i no-topology snap ${fileInfo.path.replace(
+        '.zip',
+        '',
+      )}/*.shp -clean rewind -info -o ${outputKey}`,
     );
 
     return JSON.parse(_geoJson[outputKey].toString('utf-8'));
   }
 
+  isValidGeoJson(geoJson: any): Error | void {
+    if (
+      geoJson.type !== 'FeatureCollection' ||
```

Comment:
have you tried `pure-geojson-validation`?

Reply:
I didn't, I'll have a look thanks @kgajowy !

Reply:
@kgajowy I had a look and I realised  [you use it in the API,](you use it in the) but then you 'vanilla check' same things as mine, so I'm not sure which is the point to use it if `mapshapper` does not complain at generating the GeoJSON